internal/providers/system/system.go

```diff
@@ -35,7 +38,37 @@ const (
 )
 
 func FetchBaseConfig(logger *log.Logger) (types.Config, report.Report, error) {
-	return fetchConfig(logger, baseFilename)
+	var fullBaseConfig types.Config
+	baseConfig, _, err := fetchConfig(logger, baseFilename)
+	if err != nil {
+		logger.Err("couldn't fetch base config: %v", err)
+		return types.Config{}, report.Report{}, err
+	}
+	fullBaseConfig = latest.Merge(fullBaseConfig, baseConfig)
+
+	// For merging base.d config fragments
+	baseDConfig, _, err := fetchBaseConfig(logger, "base.d")
+	if err != nil {
+		logger.Warning("no config at base.d: %v", err)
+	}
```

Comment:
I think it'd be clearer to merge the `baseDConfig` into `fullBaseConfig` immediately, rather than keeping track of which local variables need to be merged later.